main.go

```diff
@@ -1,121 +1,47 @@
 package main
 
 import (
-	"encoding/json"
-	"io/ioutil"
 	"log"
 	"net/http"
-)
 
-const (
-	getPostsEndpoint    = "https://my-json-server.typicode.com/typicode/demo/posts"
-	getCommentsEndpoint = "https://my-json-server.typicode.com/typicode/demo/comments"
+	"bathien/assignment/entity"
+	"bathien/assignment/renderer"
+	"bathien/assignment/repository"
 )
 
-type Post struct {
-	ID    int64  `json:"id"`
-	Title string `json:"title"`
-}
-type Comment struct {
-	ID     int64  `json:"id"`
-	Body   string `json:"body"`
-	PostID int64  `json:"postId"`
-}
-type PostWithCommentsResponse struct {
-	Posts []PostWithComments `json:"posts"`
-}
-type PostWithComments struct {
-	ID       int64     `json:"id"`
-	Title    string    `json:"string"`
-	Comments []Comment `json:"comments,omitempty"`
-}
-
-//TODO: how to separate API logic, business logic and response format logic
 func main() {
+	httpClient := http.DefaultClient
+	xmlRenderer := &renderer.JSONRenderer{}
+	httpRepository, httpRepositoryErr := repository.New(repository.WithHTTPClient(httpClient))
```

Comment:
Just keep an error named `err` for most case. If need specify what error is for that case then can be named like `httpRepositoryErr`